iml-warp-drive/src/main.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+// Long and nested future chains can quickly result in large generic types.
+#![type_length_limit = "16777216"]
+
+use std::{
+    collections::HashMap,
+    ops::DerefMut,
+    sync::{Arc, Mutex},
+};
+
+use futures::{
+    future::{lazy, Future},
+    sync::oneshot,
+    Stream,
+};
+use lapin_futures::client::ConnectionOptions;
+use warp::Filter;
+
+use iml_warp_drive::{
+    env,
+    locks::{self, create_locks_consumer, Locks},
+    rabbit::connect,
+    users,
+};
+
+type SharedLocks = Arc<Mutex<Locks>>;
+
+fn main() {
+    env_logger::init();
+
+    // Keep track of all connected users, key is usize, value
+    // is a event stream sender.
+    let user_state: users::SharedUsers = Arc::new(Mutex::new(HashMap::new()));
+    let lock_state: SharedLocks = Arc::new(Mutex::new(HashMap::new()));
+
+    // Clone here to allow SSE route to get a ref.
+    let user_state2 = user_state.clone();
+    let lock_state2 = lock_state.clone();
+
+    // Handle an error in locks by shutting down
+    let (tx, rx) = oneshot::channel();
+
+    let user_state3 = user_state.clone();
+    let rx = rx.inspect(move |_| users::disconnect_all_users(&user_state3));
+
+    tokio::run(lazy(move || {
+        warp::spawn(
+            connect(
+                &env::get_addr(),
+                ConnectionOptions {
+                    username: env::get_user(),
+                    password: env::get_password(),
+                    vhost: env::get_vhost(),
+                    heartbeat: 0, // Turn off heartbeat due to https://github.com/sozu-proxy/lapin/issues/152
+                    frame_max: 65535,
```

Comment:
Would it not make sense to make this a constant? I'm assuming this value is in documentation somewhere. Do you think it would be helpful to put a link to that documentation in here?